Hi @stevenmanton 

Thank you a lot for the PR! Currently the test will fail for the following 
```bash
FAILED src/transformers/models/perceiver/modeling_perceiver.py::transformers.models.perceiver.modeling_perceiver.PerceiverForImageClassificationConvProcessing.forward
FAILED src/transformers/models/perceiver/modeling_perceiver.py::transformers.models.perceiver.modeling_perceiver.PerceiverForImageClassificationFourier.forward
FAILED src/transformers/models/perceiver/modeling_perceiver.py::transformers.models.perceiver.modeling_perceiver.PerceiverForImageClassificationLearned.forward
```
We should add the expected outputs in the doc examples.
For other doc examples in this model, the tests pass, but they don't have outputs to test.
For example, `PerceiverForOpticalFlow` has
```
>>> logits = outputs.logits
```
without output and the expected value. We can have something like

```python
>>> list(logits .shape)
expected shapes
```
Would you like to fix and enhance those doc examples?

Once you have a change (staged or commited), you can run the test like
```
python utils/prepare_for_doc_test.py src/transformers/models/perceiver/modeling_perceiver.py
pytest --doctest-modules src/transformers/models/perceiver/modeling_perceiver.py -sv --doctest-continue-on-failure
```
Once the test is run, you can clean up the git status before further changes or push.

Thanks!

--- a/src/transformers/models/perceiver/modeling_perceiver.py
+++ b/src/transformers/models/perceiver/modeling_perceiver.py
@@ -1269,6 +1269,7 @@ class PerceiverForImageClassificationLearned(PerceiverPreTrainedModel):
         >>> # model predicts one of the 1000 ImageNet classes
         >>> predicted_class_idx = logits.argmax(-1).item()
         >>> print("Predicted class:", model.config.id2label[predicted_class_idx])
+        Predicted class: tabby, tabby cat
         ```"""
         if inputs is not None and pixel_values is not None:
             raise ValueError("You cannot use both `inputs` and `pixel_values`")
@@ -1406,6 +1407,7 @@ class PerceiverForImageClassificationFourier(PerceiverPreTrainedModel):
         >>> # model predicts one of the 1000 ImageNet classes
         >>> predicted_class_idx = logits.argmax(-1).item()
         >>> print("Predicted class:", model.config.id2label[predicted_class_idx])
+        Predicted class: tabby, tabby cat
         ```"""
         if inputs is not None and pixel_values is not None:
             raise ValueError("You cannot use both `inputs` and `pixel_values`")
@@ -1543,6 +1545,7 @@ class PerceiverForImageClassificationConvProcessing(PerceiverPreTrainedModel):
         >>> # model predicts one of the 1000 ImageNet classes
         >>> predicted_class_idx = logits.argmax(-1).item()
         >>> print("Predicted class:", model.config.id2label[predicted_class_idx])
+        Predicted class: tabby, tabby cat
         ```"""
         if inputs is not None and pixel_values is not None:
             raise ValueError("You cannot use both `inputs` and `pixel_values`")

--- a/src/transformers/models/perceiver/modeling_perceiver.py
+++ b/src/transformers/models/perceiver/modeling_perceiver.py
@@ -801,6 +801,8 @@ class PerceiverModel(PerceiverPreTrainedModel):
         >>> with torch.no_grad():
         ...     outputs = model(inputs=inputs)
         >>> logits = outputs.logits
+        >>> list(logits.shape)
+        [1, 2]
 
         >>> # to train, one can train the model using standard cross-entropy:
         >>> criterion = torch.nn.CrossEntropyLoss()
@@ -845,6 +847,8 @@ class PerceiverModel(PerceiverPreTrainedModel):
         >>> with torch.no_grad():
         ...     outputs = model(inputs=inputs)
         >>> logits = outputs.logits
+        >>> list(logits.shape)
+        [1, 2]
 
         >>> # to train, one can train the model using standard cross-entropy:
         >>> criterion = torch.nn.CrossEntropyLoss()
@@ -1019,6 +1023,8 @@ class PerceiverForMaskedLM(PerceiverPreTrainedModel):
         >>> outputs = model(**inputs, labels=labels)
         >>> loss = outputs.loss
         >>> logits = outputs.logits
+        >>> list(logits.shape)
+        [1, 2048, 262]
 
         >>> # inference
         >>> text = "This is an incomplete sentence where some words are missing."
@@ -1031,6 +1037,8 @@ class PerceiverForMaskedLM(PerceiverPreTrainedModel):
         >>> with torch.no_grad():
         ...     outputs = model(**encoding)
         >>> logits = outputs.logits
+        >>> list(logits.shape)
+        [1, 2048, 262]
 
         >>> masked_tokens_predictions = logits[0, 52:61].argmax(dim=-1).tolist()
         >>> tokenizer.decode(masked_tokens_predictions)
@@ -1129,6 +1137,8 @@ class PerceiverForSequenceClassification(PerceiverPreTrainedModel):
         >>> inputs = tokenizer(text, return_tensors="pt").input_ids
         >>> outputs = model(inputs=inputs)
         >>> logits = outputs.logits
+        >>> list(logits.shape)
+        [1, 2]
         ```"""
         if inputs is not None and input_ids is not None:
             raise ValueError("You cannot use both `inputs` and `input_ids`")
@@ -1266,6 +1276,8 @@ class PerceiverForImageClassificationLearned(PerceiverPreTrainedModel):
         >>> inputs = feature_extractor(images=image, return_tensors="pt").pixel_values
         >>> outputs = model(inputs=inputs)
         >>> logits = outputs.logits
+        >>> list(logits.shape)
+        [1, 1000]
         >>> # model predicts one of the 1000 ImageNet classes
         >>> predicted_class_idx = logits.argmax(-1).item()
         >>> print("Predicted class:", model.config.id2label[predicted_class_idx])
@@ -1404,6 +1416,8 @@ class PerceiverForImageClassificationFourier(PerceiverPreTrainedModel):
         >>> inputs = feature_extractor(images=image, return_tensors="pt").pixel_values
         >>> outputs = model(inputs=inputs)
         >>> logits = outputs.logits
+        >>> list(logits.shape)
+        [1, 1000]
         >>> # model predicts one of the 1000 ImageNet classes
         >>> predicted_class_idx = logits.argmax(-1).item()
         >>> print("Predicted class:", model.config.id2label[predicted_class_idx])
@@ -1542,6 +1556,8 @@ class PerceiverForImageClassificationConvProcessing(PerceiverPreTrainedModel):
         >>> inputs = feature_extractor(images=image, return_tensors="pt").pixel_values
         >>> outputs = model(inputs=inputs)
         >>> logits = outputs.logits
+        >>> list(logits.shape)
+        [1, 1000]
         >>> # model predicts one of the 1000 ImageNet classes
         >>> predicted_class_idx = logits.argmax(-1).item()
         >>> print("Predicted class:", model.config.id2label[predicted_class_idx])
@@ -1693,6 +1709,8 @@ class PerceiverForOpticalFlow(PerceiverPreTrainedModel):
         >>> patches = torch.randn(1, 2, 27, 368, 496)
         >>> outputs = model(inputs=patches)
         >>> logits = outputs.logits
+        >>> list(logits.shape)
+        [1, 368, 496, 2]
         ```"""
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
@@ -1919,6 +1937,12 @@ class PerceiverForMultimodalAutoencoding(PerceiverPreTrainedModel):
 
         >>> outputs = model(inputs=inputs, subsampled_output_points=subsampling)
         >>> logits = outputs.logits
+        >>> list(logits['audio'].shape)
+        [1, 240]
+        >>> list(logits['image'].shape)
+        [1, 6272, 3]
+        >>> list(logits['label'].shape)
+        [1, 700]
         ```"""
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 

--- a/src/transformers/models/perceiver/modeling_perceiver.py
+++ b/src/transformers/models/perceiver/modeling_perceiver.py
@@ -2953,7 +2953,6 @@ class PerceiverAudioPostprocessor(nn.Module):
         self.classifier = nn.Linear(in_channels, config.samples_per_patch)
 
     def forward(self, inputs: torch.Tensor, pos: Optional[torch.Tensor] = None, modality_sizes=None) -> torch.Tensor:
-
         logits = self.classifier(inputs)
         return torch.reshape(logits, [inputs.shape[0], -1])
 

